src/components/sbb-wagon/sbb-wagon.tsx

```diff
@@ -0,0 +1,170 @@
+import { Component, Element, Fragment, h, JSX, Prop, State } from '@stencil/core';
+import { InterfaceSbbWagonAttributes } from './sbb-wagon.custom.d';
+import getDocumentLang from '../../global/helpers/get-document-lang';
+import {
+  i18nWagonLabel,
+  i18nClass,
+  i18nOccupancy,
+  i18nBlockedPassage,
+  i18nLocomotiveLabel,
+  i18nClosedCompartmentLabel,
+} from '../../global/i18n';
+
+let nextId = 0;
+
+/**
+ * @slot unnamed - Used to slot one to x icons for meta information of the sbb-wagon.
+ */
+
+@Component({
+  shadow: true,
+  styleUrl: 'sbb-wagon.scss',
+  tag: 'sbb-wagon',
+})
+export class SbbWagon {
+  /** Wagon type. */
+  @Prop({ reflect: true }) public type: InterfaceSbbWagonAttributes['type'] = 'wagon';
+
+  /** Occupancy of a wagon. */
+  @Prop() public occupancy: InterfaceSbbWagonAttributes['occupancy'] = 'unknown';
+
+  @Prop() public blockedPassage: InterfaceSbbWagonAttributes['blockedPassage'] = 'none';
+
+  /** Visible class label of a wagon. */
+  @Prop() public wagonClass?: '1' | '2';
+
+  /** Visible label for the wagon number. Not used by type locomotive or blocked. */
+  @Prop() public label?: string;
+
+  /** Custom accessibility text to overwrite the constructed default text. */
+  @Prop() public customAccessibilityLabel = '';
+
+  /** Additional accessibility text which will be appended to the constructed default text. */
+  @Prop() public additionalAccessibilityText = '';
+
+  /** Accessibility-text for translations as the list title for additional information icons on a wagon. */
+  @Prop() public accessibilityLabelIconListTitle = '';
+
+  /** This id will be forwarded to the relevant inner element. */
+  @Prop() public iconListTitleId = `sbb-wagon-list-title-${++nextId}`;
```

Comment:
this prop can be removed as ids from outside don't work. inside this component you can use a fixed and not generated id.
```suggestion
```